pagefaultgames/pokerogue
# Implement support for relearner only moves

## What are the changes?
Implements support for relearner only moves, moves that aren't learned by level up or evolution, only by memory mushrooms.

## Why am I doing these changes?
This has been reported before, and is an outstanding bug.

## What did change?
This is mostly just engine changes to support level -1 moves as relearner exclusive moves. It also adds a few extra comments and type annotations to functions touched by this PR.

## How to test the changes?
Look at Zangoose in the starter screen and notice it no longer gets 15 moves at starter select.
You can also test that memory mushrooms still give those moves, and that level up/evolution moves still work.

## Checklist
